sympy/solvers/ode/single.py

```diff
@@ -784,5 +784,72 @@ def _get_general_solution(self, *, simplify: bool = True):
         return [gensol]
 
 
+class SecondNonlinearAutonomousConserved(SinglePatternODESolver):
+    r"""
+    Gives solution for the autonomous second order nonlinear
+    differential equation of the form
+
+    .. math :: f''(x) = g(f(x))
+
+    The solution for this differential equation can be computed
+    by multiplying by `f'(x)` and integrating on both sides,
+    converting it into a first order differential equation.
+
+    Examples
+    ========
+
+    >>> from sympy import Function, Symbol, dsolve
+    >>> f = Function('f')
+    >>> x = Symbol('x')
+
+    >>> from sympy import sqrt
+    >>> eq = f(x).diff(x, 2) + sqrt(f(x))
+    >>> dsolve(eq, simplify=False)
+    [Eq(2*f(x)*gamma(2/3)*hyper((1/2, 2/3), (5/3,), 4*exp_polar(2*I*pi)*f(x)**(3/2)/(3*C1))/(3*sqrt(C1)*gamma(5/3)) - 2*gamma(2/3)*hyper((1/2, 2/3), (5/3,), 4*exp_polar(2*I*pi)/(3*C1))/(3*sqrt(C1)*gamma(5/3)), C2 + x),
+    Eq(2*f(x)*gamma(2/3)*hyper((1/2, 2/3), (5/3,), 4*exp_polar(2*I*pi)*f(x)**(3/2)/(3*C1))/(3*sqrt(C1)*gamma(5/3)) - 2*gamma(2/3)*hyper((1/2, 2/3), (5/3,), 4*exp_polar(2*I*pi)/(3*C1))/(3*sqrt(C1)*gamma(5/3)), C2 - x)]
+
+    >>> from sympy import exp, log
+    >>> eq = f(x).diff(x, 2) - exp(f(x)) + log(f(x))
+    >>> dsolve(eq, simplify=False)
+    [Eq(Integral(1/sqrt(C1 - 2*u*log(u) + 2*u + 2*exp(u)), (u, 1, f(x))), C2 + x),
+    Eq(Integral(1/sqrt(C1 - 2*u*log(u) + 2*u + 2*exp(u)), (u, 1, f(x))), C2 - x)]
+
+    References
+    ==========
+
+    http://eqworld.ipmnet.ru/en/solutions/ode/ode0301.pdf
+    """
+    hint = "2nd_nonlinear_autonomous_conserved"
+    has_integral = True
+    order = [2]
+
+    def _wilds(self, f, x, order):
+        fy = Wild('fy', exclude=[0, f(x).diff(x), f(x).diff(x, 2)])
+        return (fy,)
+
+    def _equation(self, fx, x, order):
+        fy = self.wilds()[0]
+        return fx.diff(x, 2) + fy
+
+    def _verify(self, fx):
+        fy = self.wilds_match()[0]
+        u = Symbol('u')
+        syms = list(map(lambda sym:sym.name, fy.subs(fx, u).free_symbols))
+        if 'x' in syms:
+            return False
+        return True
+
+    def _get_general_solution(self, *, simplify: bool = True):
+        g = self.wilds_match()[0]
+        fx = self.ode_problem.func
+        x = self.ode_problem.sym
+        u = Symbol('u')
+        g = g.subs(fx, u)
+        C1, C2 = self.ode_problem.get_numbered_constants(num=2)
+        inside = -2*Integral(g, u) + C1
+        lhs = Integral(1/sqrt(inside), (u, 1, fx))
+        return [Eq(lhs, C2 + x), Eq(lhs, C2 - x)]
```

Comment:
Why does the integral go from 1?

Reply:
This is something I saw in Wolfram, the integral is calculated from 1 to `f(x)`. Should I remove it?

Reply:
Should I just make it `Integral(1/sqrt(inside), (u, fx))` instead of `Integral(1/sqrt(inside), (u, 1, fx))`?

Reply:
Yes, I think so.